### PR TITLE
Add ActiveStorage

### PR DESCRIPTION
**Pull request by kennyevil**
## Changes in this PR

Added ActiveStorage and Support::EmailAttachment model

### CODE REVIEW

**Comment by kennyevil** [comment]
[#248 - Database model for attachment storage](https://trello.com/c/uOJQd9sZ/253-248-database-model-for-attachment-storage)

**Review by AndrewCowle — APPROVED** [review]

File: config/environments/production.rb [hunk 1]
```diff
@@ -122,4 +122,7 @@
   config.action_dispatch.default_headers = {
     "X-Xss-Protection" => "0",
   }
+
+  # Set active storage location
+  config.active_storage.service = :amazon
```
Comment (ryantk) [inline config/environments/production.rb:127]: Can you put an if statement around this, we have not yet provisioned an s3 bucket for production but we will have one in preview (counts as production).

something along the lines of If AWS_S3_BUCKET_ID ? :amazon : :local

**Review by ryantk — CHANGES_REQUESTED** [review]

File: .env.example [hunk 2]
```diff
@@ -52,3 +52,10 @@ MS_GRAPH_TENANT=
 MS_GRAPH_CLIENT_ID=
 MS_GRAPH_CLIENT_SECRET=
 MS_GRAPH_SHARED_MAILBOX_USER_ID=
+
+# AWS S3
+
+AWS_S3_ACCESS_KEY_ID=
```
Comment (ryantk) [inline .env.example:58]: I need you to change these env vars too. 

The environment variables will be exposed to the application via "vcap".

We need the environment variables to match those under "credentials": https://docs.cloud.service.gov.uk/deploying_services/s3/#connect-to-an-s3-bucket-from-your-app

"aws_access_key_id" -> AWS_ACCESS_KEY_ID
 "aws_region" -> AWS_REGION 
 "aws_secret_access_key" -> AWS_SECRET_ACCESS_KEY
 "bucket_name" -> BUCKET_NAME
 
 some of those env vars may already exist in the app.
 
For context: https://github.com/DFE-Digital/buy-for-your-school/blob/253b2719359e63abb135bee386d522cc8341b134/lib/vcap_parser.rb#L14

**Review by ryantk — CHANGES_REQUESTED** [review]

File: lib/vcap_parser.rb [hunk 3]
```diff
@@ -26,4 +30,13 @@ def self.load_redis_config(redis_config)
     # Generate a REDIS_URL from the redis service uri
     ENV["REDIS_URL"] = redis_config.fetch("credentials").fetch("uri")
   end
+
+  def self.load_s3_config(s3_config)
+    return unless s3_config
+
+    # set S3 credentials
+    s3_config.fetch("credentials").each_pair do |key, value|
+      ENV[key.capitalize] = value
```
Comment (ryantk) [inline lib/vcap_parser.rb:39]: Change `capitalize` to `upcase` it won't work quite as imagined, see below:

```
irb(main):008:0> "aws_secret_access_key".capitalize
=> "Aws_secret_access_key"
irb(main):009:0> "aws_secret_access_key".upcase
=> "AWS_SECRET_ACCESS_KEY"
```

**Review by ryantk — CHANGES_REQUESTED** [review]

**Pull request merged**